src/Avalonia.Input/MouseDevice.cs

```diff
@@ -147,6 +147,8 @@ private void ProcessRawEvent(RawPointerEventArgs e)
             if(mouse._disposed)
                 return;
 
+            if (e.Type == RawPointerEventType.NonClientLeftButtonDown) return;
+
             _position = e.Root.PointToScreen(e.Position);
```

Comment:
This position is stored and used as a parameter for SetPointerOver calls here
https://github.com/AvaloniaUI/Avalonia/blob/ee648c7c2c76cb9bcdf3b0833890aeafe30ea083/src/Avalonia.Input/MouseDevice.cs#L109-L123
This position is not correct for `RawPointerEventType.NonClientLeftButtonDown`-s though.

Reply:
Position **was** correct at click time but due to resizing it cached a stale value. But anyway this old logic was conflicting with window pointer leave mechanic (which was clearing pointerover state) so your fix looks fine to me.